### DISCUSSION TITLE
常见问题汇总

### DISCUSSION THREAD

**Discussion by Yin-Hongwei** [Yin-Hongwei/vue-spring-music#182] [Q&A]
### 1、图片、音乐加载失败

把 music-website/music-server 目录下的 img、song 目录移动到 music-website 目录。**【拉的最新的代码可忽略】**

### 2、音乐播放不了

可能是音乐损毁了，重新更换一下音乐资源。

**Comment by 1-yang** [comment]
图片下载的提取码是错的

**Reply by Yin-Hongwei** [reply]
链接: https://pan.baidu.com/s/1ZW230mb0MlDHzyceNKMcIQ 提取码: vst9

**Comment by Kathrsis** [comment]


**Reply by Yin-Hongwei** [reply]
> 大佬下载功能怎么下载下来的音乐是一个空文件？

你看看是不是你音乐损毁了，不应该

**Reply by Jade-time** [reply]
音乐在网页上能播放，存放的文件也没问题，但是下载下来后就出错了

**Reply by CosmicCharmer** [reply]
同问

**Reply by 424670683** [reply]
我也是这里出问题

**Comment by pangpee** [comment]
​ 搜索中文搜不出来，改了一下数据库把歌名改成数字，就能搜索出来，咋解决呀​
![搜索汉字](https://github.com/Yin-Hongwei/music-website/assets/134503507/b7d3e7b5-6507-4b2c-b7df-293f9461206d)
![搜索3](https://github.com/Yin-Hongwei/music-website/assets/134503507/242bfc96-bfb7-4b69-a2b3-b568b403fb95)

**Reply by pangpee** [reply]
轮播图显示不出来，但其他图片都能正常显示

**Reply by Yin-Hongwei** [reply]
> 轮播图显示不出来，但其他图片都能正常显示

轮播图问题看readme，需要起redis，搜索问题建议数据库版本5.7.21以上的

**Reply by HengZan** [reply]
application-dev.properties   加一句characterEncoding=utf-8

**Reply by shibaemail** [reply]
你好我也是这个问题  请问解决了吗

**Comment by JimBLuesLin** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/102970295/c0496c30-2dca-49c5-9bc6-9fba88a4f6fb)
这样是因为data的路径没配好吗

**Reply by Yin-Hongwei** [reply]
data没配置只是图片音乐看不了，报错是404，有500是服务端有问题，整个界面截图吧

**Reply by P628GO** [reply]
哥们你解决了吗 啥原因啊

**Comment by JimBLuesLin** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/102970295/67fd4b00-10ff-4df9-a34c-926960fbc457)
页面是这样的

**Reply by Yin-Hongwei** [reply]
后端没启动后数据库没连接

**Comment by ihianian** [comment]


**Reply by ihianian** [reply]
哦哦，没问题了，我看到拦截器中有地址转换。

**Comment by dyk-npu** [comment]


**Reply by dyk-npu** [reply]
RankListMapper中selectUserRank返回类型int，账号进入没有打过分的歌单会在页面上出现Uncaught runtime errors，好像是没打分的歌单这个值是null，int不允许null，然后报错了，我将数据类型修改成了Integer但是每次进入都是变成了已打分，打的是0分。

**Reply by Yin-Hongwei** [reply]
这是个bug，还没看😂

**Comment by Goodusna** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/86639043/a5327e54-3133-47c5-b4af-059a8f57df81)
提示这种没有走缓存怎么解决啊，readits都启动了，也可以ping通

**Comment by yuluo-yx** [comment]
127.0.0.1/:18     GET http://localhost:8888//song/%E8%89%BA%E5%A3%B0-%E1%84%86%E1%85%AE%E1%86%AB%20%E1%84%8B%E1%85%A7%E1%86%AF%E1%84%8B%E1%85%A5%E1%84%87%E1%85%AA%20(Here%20I%20am).mp3 404

后端没有这个接口？

org.apache.ibatis.binding.BindingException: Mapper method 'com.example.yin.mapper.RankListMapper.selectUserRank attempted to return null from a method with a primitive return type (int).
	at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.execute(MybatisMapperMethod.java:105)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy$PlainMethodInvoker.invoke(MybatisMapperProxy.java:148)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy.invoke(MybatisMapperProxy.java:89)
	at com.sun.proxy.$Proxy89.selectUserRank(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:137)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)


还有这问题？项目确定可以跑通吗？

**Comment by dyk-npu** [comment]
没有，好像就是数据类型的问题



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年8月21日(周一) 下午3:24
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182)




 
哥们你解决了吗 啥原因啊

**Comment by P628GO** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/82953786/67595cee-736b-42b5-b309-8631e0965bfa)
xd我这个是资源文件出错了吗？报500是服务器有问题哪里代码没改好吗

**Comment by dyk-npu** [comment]
没显示图片和歌单的话应该是资源路径的问题，你看看数据库表里的资源相对路径



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年8月21日(周一) 下午3:51
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182)




 

 xd我这个是资源文件出错了吗？报500是服务器有问题哪里代码没改好吗

**Reply by P628GO** [reply]
哥 我看了那个后端的报错 acess deny jdbc connection。但是我的application.properties里相关配置都改了，也不知道为啥还连不上：（

**Comment by P628GO** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/82953786/025253d6-8ea5-4e5c-92ec-fef26ba84a92)
用户名和密码都是对的，但是我把上面那个地址改成localhost了

**Comment by dyk-npu** [comment]
localhost和127.0.0.1是一个意思，你如果用的是mysql8以上的链接要写com.mysql.cj.jdbc.Driver



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年8月21日(周一) 下午4:34
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182)




 
用户名是对的 密码也是对的 我还把上面那个地址改成localhost了（难道不用改吗）

**Reply by P628GO** [reply]
哥 我解决了 啊啊啊啊我才看到有俩配置文件 我只修改了一个 把俩都改了就正常显示了。谢谢哥

**Comment by dyk-npu** [comment]
我搞了一下后，打分功能变正常了，不知道是不是vue版本的问题



---原始邮件---
发件人: "Wang ***@***.***&gt;
发送时间: 2023年8月31日(周四) 下午5:31
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182)




 
好兄弟能加个联系方式吗 我跟你一模一样的报错 还没改好

**Reply by dyk-npu** [reply]
我想起来了，前端SongSheetDetail.vue里改一下getUserRank函数就行了，然后后端将selectUserRank数据的返回类型改为Integer
![1](https://github.com/Yin-Hongwei/music-website/assets/76047834/ab6c909a-aebe-41e6-8a47-7fd4733782d3)

**Comment by dyk-npu** [comment]
我也不是很懂啊



---原始邮件---
发件人: "Wang ***@***.***&gt;
发送时间: 2023年8月31日(周四) 下午5:53
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182)




 
大佬能加个联系方式吗 我本地运行有些问题想请教

**Comment by dingfhua** [comment]
music-client执行npm install   npm run dev 启动不了啊，music-manager可以正常启动

**Reply by xzsx666** [reply]
一堆问题。都不知道怎么打开网站。

**Comment by xzsx666** [comment]
启动服务端，方法一命令应该是.\mvnw spring-boot:run

**Comment by maye-zc** [comment]
用户能够注册但不能够登录，轮播图也没了
![image](https://github.com/Yin-Hongwei/music-website/assets/147402023/9d33aa86-8cd6-4093-8d93-d27e7727567e)

**Reply by maye-zc** [reply]
轮播图也没了

**Comment by luodahunzi2505** [comment]
客户端可以正常启动，但是服务端好像不太像，说缺少文件啥的



 ERROR  Failed to compile with 1 error                                                                                                                                          09:31:16

 error 

Module not found: Error: Can't resolve './src/main.ts' in 'D:\Desktop\UPUP\music-website\music-server'

ERROR in app
Module not found: Error: Can't resolve './src/main.ts' in 'D:\Desktop\UPUP\music-website\music-server'

webpack compiled with 1 error
ERROR in tsconfig.json
TS18003: No inputs were found in config file 'tsconfig.json'. Specified 'include' paths were '["src/**/*.ts","src/**/*.tsx","src/**/*.vue","tests/**/*.ts","tests/**/*.tsx"]' and 'exclu
de' paths were '["node_modules"]'.

**Comment by yehhtt** [comment]
怎么切换到vue2呀？有没有大哥教教/?

**Comment by ayanyanyanyan** [comment]
后端打成jar包后无法运行，显示没有主清单属性，有没有老哥有办法解决以下啊‘

**Comment by tank-w** [comment]
大哥，请教一下，这个歌词是怎么获取的，我在酷狗上弄的格式后缀是krc   打开乱码

**Comment by fultur** [comment]


**Reply by Lihh7** [reply]
数据库命名要是 `tp_music` ，然后 那两个`spring.datasource.password` 密码都改成自己的密码

**Comment by lhhy2019** [comment]
![_)MX$4ZF$JCYLIXHN`2Q9EB](https://github.com/Yin-Hongwei/music-website/assets/151428339/1d6c5c4c-dbca-4047-a5c8-f74ecaf0c9e3)
求，这咋回事啊，没有图片，音乐也播放不了

**Reply by lhhy2019** [reply]
已解决，原因是没按照这个文件的放置方法

![image](https://github.com/Yin-Hongwei/music-website/assets/151428339/92b83296-a17e-4b47-8264-26d02125d0eb)
![Uploading image.png…]()

**Reply by YNPLMS** [reply]
怎么解决，路径还是没有解决

**Reply by lhhy2019** [reply]
报404说明是图片的路径找不到，原因是文件夹对应的文件没放对，我是这样放的
![image](https://github.com/Yin-Hongwei/music-website/assets/151428339/ede0a648-9044-40af-ac5a-1feb7b6c29c4)

**Comment by Apollo-l** [comment]
有老哥遇到这个问题吗，怎么解决啊
Exception in thread "main" java.util.zip.ZipException: error in opening zip file
        at java.util.zip.ZipFile.open(Native Method)
        at java.util.zip.ZipFile.<init>(ZipFile.java:225)
        at java.util.zip.ZipFile.<init>(ZipFile.java:155)
        at java.util.zip.ZipFile.<init>(ZipFile.java:169)
        at org.apache.maven.wrapper.Installer.unzip(Installer.java:161)
        at org.apache.maven.wrapper.Installer.createDist(Installer.java:78)
        at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:121)
        at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:55)
![图片1](https://github.com/Yin-Hongwei/music-website/assets/82933233/82d01966-a9b3-4d2d-8a90-556d0779661d)

**Comment by use8879576** [comment]
您好， mysql 中 table : song 的字段 lyric的信息 在前端哪里展示？

**Comment by wht15116532585** [comment]
![9e25e9218d40cc031b721fbde7641fe9](https://github.com/Yin-Hongwei/music-website/assets/163106654/8bca7566-6d46-4ac3-951d-3f413257f1a4)
已经把 music-website/music-server 目录下的 img、song 目录移动到 music-website 目录下，但还是加载不出来

**Reply by xzsx666** [reply]
我都加载出来了

**Reply by xzsx666** [reply]
三个文件要分开

**Comment by xzsx666** [comment]
网站正常显示，用的数据库账号密码，就是登录不了，显示密码错误咋回事？？？

**Reply by wht15116532585** [reply]
应该是服务器没启动

**Reply by Yin-Hongwei** [reply]
数据库里的密码是加密的，不能用来直接登录，自己重注册一个吧

**Comment by wht15116532585** [comment]


**Reply by ssk1tlz** [reply]
告诉我更多关于你在哪里放了什么。

**Comment by Jade-time** [comment]
后台管理员账户可以在数据库录入使用，但是前台用户的密码录入数据库会经过加密，所以看用户帐号表的密码是一长串的，如果觉得自行加密麻烦，也可以在系统注册，这样就能用



---- Replied Message ----
| From | ***@***.***> |
| Date | 03/15/2024 17:39 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182) |

网站正常显示，用的数据库账号密码，就是登录不了，显示密码错误咋回事？？？

**Reply by ssk1tlz** [reply]
您能帮我建立数据库连接吗？

**Reply by Yin-Hongwei** [reply]
1. 本地安装 mysql   🫱 根据你使用的是windows还是mac系统找个教程配置一下，安装完成后启动 mysql
2. 安装 mysqlWorkbench  🫱 https://www.mysql.com/products/workbench/
3. 使用 mysqlWorkbench 导入项目中的 sql 文件
4. 在项目中配置用户和密码链接你的数据库

**Comment by thebestBaoBao** [comment]
用户头像加载不出来怎么办？已经放了一张user图片了
<img width="550" alt="image" src="https://github.com/Yin-Hongwei/music-website/assets/96944461/dd1be9ec-2ae8-4cb2-95d0-c391fd645555">

**Reply by Yin-Hongwei** [reply]
用户头像可以编辑，重新上传一张吧

**Reply by thebestBaoBao** [reply]
头像上传失败，控制台没报错

**Reply by Yin-Hongwei** [reply]
需要看控制台的networkhuo zhe或者后端的控制台
![image](https://github.com/Yin-Hongwei/music-website/assets/37996435/e1978fa0-2e09-4c4c-85ea-c52583ccd263)

**Reply by thebestBaoBao** [reply]
![屏幕截图 2024-03-27 205449](https://github.com/Yin-Hongwei/music-website/assets/96944461/7e8fc864-24ab-427a-a80a-3a56bcfb2b19)

**Reply by thebestBaoBao** [reply]
![屏幕截图 2024-03-27 211002](https://github.com/Yin-Hongwei/music-website/assets/96944461/9a9bf084-970e-4188-96dd-04c1d66e7378)

**Reply by thebestBaoBao** [reply]
![屏幕截图 2024-03-27 211706](https://github.com/Yin-Hongwei/music-website/assets/96944461/6314d9b0-ce87-4f99-881c-1c6e71867e9c)

**Reply by Yin-Hongwei** [reply]
截图一下你资源的位置
<img width="228" alt="image" src="https://github.com/Yin-Hongwei/music-website/assets/37996435/0244fd78-55d3-44d5-af7b-de0162ce3527">

**Reply by thebestBaoBao** [reply]
![屏幕截图 2024-03-28 115147](https://github.com/Yin-Hongwei/music-website/assets/96944461/07e63834-ca11-4c2a-9b8a-1a060dab2d9d)

**Reply by wizradh** [reply]
解决了吗，我和你一样

**Comment by Yin-Hongwei** [comment]


**Reply by thebestBaoBao** [reply]
能的

**Comment by hsjy** [comment]
大佬能否帮我看下这是什么问题？
![image](https://github.com/Yin-Hongwei/music-website/assets/113044703/d9cc4888-964b-4b01-93ff-59e456f8f89f)
Cannot read properties of undefined (reading 'status')
TypeError: Cannot read properties of undefined (reading 'status')
    at eval (webpack-internal:///./src/api/request.ts:32:22)

**Reply by Yin-Hongwei** [reply]
可能是你node版本有点问题，换到14.17.3，项目重install一下，再dev试试

**Comment by boaiyixiao** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/163430712/13e0d5d5-df64-44a4-9914-cfa20cee01ea)
![微信图片_20240330160846](https://github.com/Yin-Hongwei/music-website/assets/163430712/c036062d-5eb0-4251-a089-0f9c7ee8f96b)
启动后台出现以上问题，是什么原因，大佬求指教

**Comment by boaiyixiao** [comment]
![Uploading 微信图片_20240330224026.png…]()

> 这个是什么原因，有部分歌不能听，还有部分图片加载不出来

**Comment by ssk1tlz** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/4a422757-90bf-45d9-80e7-a981fa6fec36)
这个错误有什么问题？当我点击音乐图标时，一切正常，这个错误是怎么回事？

**Reply by NogharaBill** [reply]
我也是，不知道为什么，一点首页或者点音乐图标就会弹出来，但是不影响功能使用

**Reply by ssk1tlz** [reply]
播放音乐无法正常工作可能是一个错误吗？

**Reply by ssk1tlz** [reply]
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/983082ba-9407-4e6c-b9e5-def77ded77fd)
我也遇到了这个错误

**Reply by boaiyixiao** [reply]
我也是这个错误，歌都能听，就是一点图标就报这个错误

**Reply by ssk1tlz** [reply]
谢谢您的反馈

**Comment by hsjy** [comment]
想问一下首页、歌单、歌手页都能正常显示，但是登录和注册页报错是为什么？
![image](https://github.com/Yin-Hongwei/music-website/assets/113044703/ed854873-8966-4d39-9908-06338c26995c)
![image](https://github.com/Yin-Hongwei/music-website/assets/113044703/d77f7053-1cd7-4a22-b95b-db5dbd17cb24)
恳请大佬指点一二！

**Comment by ssk1tlz** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/6c579d61-dec2-4413-b18b-9aa99ccc1bc8)
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/1ffa0660-a4ce-4abd-af9f-2ce9712029ae)
我无法添加音乐，问题出在哪里？

**Comment by shibaemail** [comment]
求解答，我的搜索功能也是只能搜到数字，搜索不到中文，有什么解决办法吗

**Reply by ssk1tlz** [reply]
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/37a51f20-f043-4102-bb4a-3f1862c1cd91)
我不知道你怎么想，但我用中英文都能正常使用。
也许你改了功能名称，顺便问一下，添加音乐对你来说效果如何？ 我在这方面遇到了问题。

**Reply by shibaemail** [reply]
我的添加功能是正常的，搜索出了问题。它不能添加和已有歌曲同一个文件名的歌曲

**Reply by ssk1tlz** [reply]
说到搜索引擎功能，我几乎帮不上忙，但说到添加网站上还没有的新歌曲，也许你能帮上忙。

**Reply by wizradh** [reply]
我搜索也是可以的，你添加功能没问题吗，能看看你minio哪里吗，我添加功能有问题

**Reply by ssk1tlz** [reply]
你知道，我没有更改任何功能，唯一做的就是更改界面语言，但我在图片和音乐方面遇到了同样的问题。

**Comment by ssk1tlz** [comment]
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/52c8281f-bc58-4816-a78b-ff232e34b862)

我在向网站添加音乐时仍然遇到困难，请帮助我!!!!！

**Reply by wizradh** [reply]
我也是这个问题，上传图片也不行，上传好像都不行

**Reply by ssk1tlz** [reply]
是的，我认为添加音乐和上传照片失灵或损坏了，我正在尝试解决这个问题，而且我遇到了很多错误，不知道如何解决。

**Comment by dingfhua** [comment]
能不能别再给我发了



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年04月05日 11:50 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [Yin-Hongwei/music-website] 常见问题汇总 (Discussion #182) |

我搜索也是可以的，你添加功能没问题吗，能看看你minio哪里吗，我添加功能有问题

**Reply by Yin-Hongwei** [reply]
取消订阅就行了 https://github.com/Yin-Hongwei/music-website/discussions/182
![image](https://github.com/Yin-Hongwei/music-website/assets/37996435/9b6a736a-d898-49ab-bdd6-10f5e944cb1b)

**Comment by yuluo-yx** [comment]
if a good solve way, please closed.

**Comment by ssk1tlz** [comment]
我再次下载了该项目，并注意到从管理面板添加音乐的功能尚未最终完成，它用红色突出显示了该数据，这是什么原因呢？
也许我不明白添加音乐的逻辑？
如果您对此有疑问，那么当我在管理面板中指定音乐的 Url 时，音乐是否被添加到了数据库中，是的，它被添加到了数据库中，但没有被添加到网站中，它也被移动到了位于音乐服务器目录下的歌曲文件夹中，作者添加的其他音乐也运行正常，也许我需要使用 Sql 查询将音乐手动添加到数据库中。
![image](https://github.com/Yin-Hongwei/music-website/assets/102754429/49f0028f-2161-4530-9434-16501b5019f9)
